rllib/offline/estimators/off_policy_estimator.py

```diff
@@ -91,6 +192,21 @@ def check_action_prob_in_batch(self, batch: SampleBatchType) -> None:
                 "`off_policy_estimation_methods: {}` to disable estimation."
             )
 
+    def _compute_action_probs(self, batch: SampleBatch):
+        log_likelihoods = compute_log_likelihoods_from_input_dict(self.policy, batch)
+        new_prob = np.exp(convert_to_numpy(log_likelihoods))
+
+        if self.epsilon_greedy > 0.0:
+            if not hasattr(self.policy.action_space, "n"):
```

Comment:
``issubclass(self.policy.action_space, (Discrete, MultiDiscrete))``?

Reply:
I don't want this to depend on those classes. The minimum abstraction needs is that it should have an `n` attribute. This is more flexible for down the line.

Reply:
hmm, is this a public spec though?
a general principle is not to rely on internal implementation details. for example, MultiBinary also has member variable named n ...
also, I kind of feel like we need to check for policy.action_space. original_space first, something like that, in case people are doing action space flattening?

Reply:
yeah, not ideal, but it seems to be everywhere already, so probably fine:
https://github.com/ray-project/ray/blob/master/rllib/algorithms/appo/appo_tf_policy.py#L167